### DISCUSSION TITLE
Can Renovate "remember" closed PRs when not run 24/7?

### DISCUSSION THREAD

**Discussion by Andre601** [renovatebot/renovate#24536] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.96.8

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate remembering past PRs it made and that got closed unmerged to not re-create the same PR again.

### What you tried so far.

This is a question rather than an issue, which this discussion template seems not fit for tbh.

My question is, if Renovate can remember past PRs it made, i.e. by using the API of the git hosting it is used on, so it won't create a PR for a dependency Bump that was rejected (Closed unmerged).

I ask this, because I use Renovate in a way that doesn't allow it to be online/active 24/7 nor having permanent logs available, as it is run as part of a CI with a cron-job.

Because of that do I fear that PRs I close may be forgotton by Renovate in the next cycle, resulting in the bot creating a new PR for the same dependency bump.

If Renovate doesn't remember past PRs like that, is there a way to make it save and commit a sort of "memory" file or similar, so it knows past dependency bumps and what PRs it made them in? So that it could then find and check the result before making another attempt.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
renovate detects closed PR's
